### ISSUE TITLE
NullPointerException: Attempt to invoke virtual method 'long com.instabug.chat.a$f.f()' on a null object reference

### ISSUE THREAD

**Issue by francos** [Instabug/Instabug-Android#265]
### Steps to Reproduce the Problem

I received this crash in Crashlytics so not sure how to reproduce it.

Stack trace:

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'long com.instabug.chat.a$f.f()' on a null object reference
       at com.instabug.chat.a.d$a.a(d.java:359)
       at com.instabug.chat.a.d$a.compare(d.java:339)
       at java.util.TimSort.countRunAndMakeAscending(TimSort.java:360)
       at java.util.TimSort.sort(TimSort.java:234)
       at java.util.Arrays.sort(Arrays.java:1498)
       at java.util.ArrayList.sort(ArrayList.java:1470)
       at java.util.Collections.sort(Collections.java:201)
       at com.instabug.chat.a.b.h(b.java:202)
       at com.instabug.chat.a.b.j(b.java:233)
       at com.instabug.chat.ui.c.a.a(a.java:114)
       at com.instabug.chat.ui.c.a.getView(a.java:70)
       at android.widget.AbsListView.obtainView(AbsListView.java:2451)
       at android.widget.HwAbsListView.obtainView(HwAbsListView.java:935)
       at android.widget.ListView.makeAndAddView(ListView.java:2078)
       at android.widget.ListView.fillSpecific(ListView.java:1500)
       at android.widget.ListView.layoutChildren(ListView.java:1808)
       at android.widget.AbsListView.onLayout(AbsListView.java:2249)
       at android.widget.HwAbsListView.onLayout(HwAbsListView.java:345)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1842)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1686)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1595)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1083)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1842)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1686)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1595)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1842)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1686)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1595)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:325)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at com.android.internal.policy.DecorView.onLayout(DecorView.java:926)
       at android.view.View.layout(View.java:20960)
       at android.view.ViewGroup.layout(ViewGroup.java:6279)
       at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3037)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2545)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1636)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7946)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1092)
       at android.view.Choreographer.doCallbacks(Choreographer.java:893)
       at android.view.Choreographer.doFrame(Choreographer.java:812)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1078)
       at android.os.Handler.handleCallback(Handler.java:907)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:216)
       at android.app.ActivityThread.main(ActivityThread.java:7625)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
```

### Expected Behavior

The app doesn't crash.

### Actual Behavior

The app crashes because of the NPE thrown in the Instabug SDK.

### Instabug integration code

```kotlin
implementation("com.instabug.library:instabug:8.7.0")
implementation("com.instabug.library:instabug-with-okhttp-interceptor:8.7.0")
```

```kotlin
Instabug.Builder(app, INSTABUG_TOKEN)
                .setInvocationEvents(SHAKE)
                .build()

Instabug.setWelcomeMessageState(WelcomeMessage.State.DISABLED)
```

### SDK Version

8.7.0

### Android Version

9

### Device Model

 P20 lite

**Comment by amrbrkt** [comment]
Thanks @FrancoSabadini for reporting this. We will look into it and come back to you ASAP. 
Would you please let me know how frequent this crash is?

**Comment by FrancoSabadini** [comment]
It only happened once so far, it happened 3 days ago.

**Comment by mzelzoghbi** [comment]
Hi @FrancoSabadini  ,

First of all, please accept our sincerest apologies for taking some time to come up with a fix for your issue. 

We created a **SNAPSHOT** that should fix the issue, Can you give it a try & let us know if it fixed the issue for you?
```
implementation 'com.instabug.library:instabug:9.0.5.3-SNAPSHOT'
```
you will need to add these lines to your app **build.gradle** before your dependencies
```
repositories {
    ....
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
}
```
This **SNAPSHOT** is completely safe to use and has no debug code at all.

Thank You 😊

**Comment by FrancoSabadini** [comment]
Hi @mzelzoghbi, it'd be hard to tell if this fixes it as I couldn't really reproduce the issue locally. Do you know if this will be included in the next release? I prefer to use normal releases rather than snapshots if possible.

**Comment by Muhammadyoussef** [comment]
Hi @francos,

This issue got fixed on `9.1.0`. It's now stable.

To consume the changes, please add this to your dependencies

`implementation 'com.instabug.library:instabug:9.1.0'`
